src/compile/selection/project.ts

```diff
@@ -63,15 +63,15 @@ const project: SelectionCompiler = {
     };
 
     const type = selCmpt.type;
-    const cfg = model.config.selection[type];
+    const cfg = model.config.selection[type] as PointSelectionConfig;
```

Comment:
What is `selCmpt.type` here? If it's telling us the type, do we still need `as PointSelectionConfig`?

Reply:
`selCmpt.type` is either `point` or `interval`. But, now, `fields` is only a property on the former, and `encodings` is on both. So, doing the typecast here felt like the cleanest fix to me rather than adding duplicative code simply for a `selCmpt.type === 'point'` type guard.

Reply:
Alternatively, I can put this typecast in at [L102](https://github.com/vega/vega-lite/pull/7442/files#diff-87e9e2677b96aab8a505667f2d85f1da7d3d205da7c9596524aa67f45ad9d11cR102):

```diff
-fields = cfg.fields;
+fields = (cfg as PointSelectionConfig).fields;
```

Let me know if that's preferable.

Reply:
I think it's best not to assert a type when we don't know it yet. I changed it in 3987a99fb.